{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [],
   "source": [
    "from Bio import AlignIO\n",
    "\n",
    "import pandas as pd\n",
    "from os.path import abspath, join\n",
    "import copy\n",
    "\n",
    "from skbio import TreeNode"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 41,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "'/panfs/panfs1.ucsd.edu/panscratch/jhc103/VertMetaphlan-frmerged/strain-trees-output/output_50_E_coli'"
      ]
     },
     "execution_count": 41,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "bacteria=\"Escherichia_coli\"\n",
    "marker_threshold = \"50\"\n",
    "sample_group=\"\" ##mammallian, nonMammallian ... \n",
    "bacteria_trimmed =bacteria[0]+bacteria[bacteria.find(\"_\"):len(bacteria)]\n",
    "output_dir=join(tree_dir,'output_'+marker_threshold+\"_\"+bacteria_trimmed + sample_group)\n",
    "output_dir"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "metadata": {},
   "outputs": [],
   "source": [
    "## Read in metadata\n",
    "md_dir = '/panfs/panfs1.ucsd.edu/panscratch/jhc103/VertMetaphlan-frmerged/metadata'\n",
    "host_md_fp = join(md_dir, 'vert_metadata_8_23.txt')\n",
    "host_md_orig = pd.read_csv(host_md_fp, sep='\\t', encoding='windows-1252')\n",
    "\n",
    "## various paths\n",
    "tree_dir = abspath('/panfs/panfs1.ucsd.edu/panscratch/jhc103/VertMetaphlan-frmerged/strain-trees-output/')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 30,
   "metadata": {},
   "outputs": [],
   "source": [
    "## Read in alignment\n",
    "align = AlignIO.read(join(output_dir,\"s__\"+bacteria+\".StrainPhlAn3_concatenated.aln\"),\"fasta\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 31,
   "metadata": {},
   "outputs": [],
   "source": [
    "## Change value of alignment\n",
    "for value in align:\n",
    "    if len(host_md_orig.loc[host_md_orig[\"SampleID2\"]==value.id,\"reads_per_file\"])==0: ## for reference genome \n",
    "        value.id = \"ref_seq\"\n",
    "    else:\n",
    "        value.id=str(int(host_md_orig.loc[host_md_orig[\"SampleID2\"]==value.id,\"reads_per_file\"]))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 32,
   "metadata": {},
   "outputs": [],
   "source": [
    "## Write new alignment \n",
    "count = AlignIO.write(align,join(output_dir,\"align.phy\"),\"phylip\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 33,
   "metadata": {},
   "outputs": [],
   "source": [
    "## Read in tree\n",
    "tree_parse= TreeNode.read(join(output_dir,\"RAxML_parsimonyTree.s__\"+bacteria+\".StrainPhlAn3.tre\"), format='newick', convert_underscores=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 34,
   "metadata": {},
   "outputs": [],
   "source": [
    "## Change tip name\n",
    "for node in tree_parse.tips():\n",
    "    if len(host_md_orig.loc[host_md_orig[\"SampleID2\"]==node.name,\"reads_per_file\"])==0: ## for reference genome \n",
    "        node.name = \"ref_seq\"\n",
    "    else:\n",
    "        node.name=str(int(host_md_orig.loc[host_md_orig[\"SampleID2\"]==node.name,\"reads_per_file\"]))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 42,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "'/panfs/panfs1.ucsd.edu/panscratch/jhc103/VertMetaphlan-frmerged/strain-trees-output/output_50_E_coli/tree_parsimony_tips_changed.nwk'"
      ]
     },
     "execution_count": 42,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "## Write tree to file\n",
    "tree_parse.write(join(output_dir,\"tree_parsimony_tips_changed.nwk\"), format=\"newick\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 44,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "align.phy\n",
      "Escherichia_coli_sample_marker_percent.tsv\n",
      "GAMMA\n",
      "RAxML_bestTree.s__Escherichia_coli.StrainPhlAn3.tre\n",
      "RAxML_info.s__Escherichia_coli.StrainPhlAn3.tre\n",
      "RAxML_log.s__Escherichia_coli.StrainPhlAn3.tre\n",
      "RAxML_parsimonyTree.s__Escherichia_coli.StrainPhlAn3.tre\n",
      "RAxML_result.s__Escherichia_coli.StrainPhlAn3.tre\n",
      "s__Escherichia_coli.info\n",
      "s__Escherichia_coli.polymorphic\n",
      "s__Escherichia_coli.StrainPhlAn3_concatenated.aln\n",
      "tree_parsimony_tips_changed.nwk\n",
      "/panfs/panfs1.ucsd.edu/panscratch/jhc103/VertMetaphlan-frmerged/strain-trees-output/output_50_E_coli/align.phy\n"
     ]
    }
   ],
   "source": [
    "!cd $output_dir\n",
    "!ls $output_dir\n",
    "!ls $output_dir/align.phy"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 47,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Warning, you specified a working directory via \"-w\"\n",
      "Keep in mind that RAxML only accepts absolute path names, not relative ones!\n",
      "\n",
      "\n",
      "Using BFGS method to optimize GTR rate parameters, to disable this specify \"--no-bfgs\" \n",
      "\n",
      "\n",
      "This is the RAxML Master Pthread\n",
      "\n",
      "This is RAxML Worker Pthread Number: 2\n",
      "\n",
      "This is RAxML Worker Pthread Number: 3\n",
      "\n",
      "This is RAxML Worker Pthread Number: 1\n",
      "\n",
      "This is RAxML Worker Pthread Number: 4\n",
      "\n",
      "This is RAxML Worker Pthread Number: 5\n",
      "\n",
      "This is RAxML Worker Pthread Number: 11\n",
      "\n",
      "This is RAxML Worker Pthread Number: 15\n",
      "\n",
      "This is RAxML Worker Pthread Number: 9\n",
      "\n",
      "This is RAxML Worker Pthread Number: 10\n",
      "\n",
      "This is RAxML Worker Pthread Number: 8\n",
      "\n",
      "This is RAxML Worker Pthread Number: 6\n",
      "\n",
      "This is RAxML Worker Pthread Number: 7\n",
      "\n",
      "This is RAxML Worker Pthread Number: 12\n",
      "\n",
      "This is RAxML Worker Pthread Number: 13\n",
      "\n",
      "This is RAxML Worker Pthread Number: 14\n",
      "\n",
      "\n",
      "This is RAxML version 8.2.12 released by Alexandros Stamatakis on May 2018.\n",
      "\n",
      "With greatly appreciated code contributions by:\n",
      "Andre Aberer      (HITS)\n",
      "Simon Berger      (HITS)\n",
      "Alexey Kozlov     (HITS)\n",
      "Kassian Kobert    (HITS)\n",
      "David Dao         (KIT and HITS)\n",
      "Sarah Lutteropp   (KIT and HITS)\n",
      "Nick Pattengale   (Sandia)\n",
      "Wayne Pfeiffer    (SDSC)\n",
      "Akifumi S. Tanabe (NRIFS)\n",
      "Charlie Taylor    (UF)\n",
      "\n",
      "\n",
      "Alignment has 5385 distinct alignment patterns\n",
      "\n",
      "Proportion of gaps and completely undetermined characters in this alignment: 28.38%\n",
      "\n",
      "RAxML Computation of pairwise distances\n",
      "\n",
      "Using 1 distinct models/data partitions with joint branch length optimization\n",
      "\n",
      "\n",
      "All free model parameters will be estimated by RAxML\n",
      "GAMMA model of rate heterogeneity, ML estimate of alpha-parameter\n",
      "\n",
      "GAMMA Model parameters will be estimated up to an accuracy of 0.1000000000 Log Likelihood units\n",
      "\n",
      "Partition: 0\n",
      "Alignment Patterns: 5385\n",
      "Name: No Name Provided\n",
      "DataType: DNA\n",
      "Substitution Matrix: GTR\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "RAxML was called as follows:\n",
      "\n",
      "/home/jhc103/miniconda3/envs/mpa3-py37/bin/raxmlHPC-PTHREADS-SSE3 -T 16 -f x -m GTRGAMMA -s /panfs/panfs1.ucsd.edu/panscratch/jhc103/VertMetaphlan-frmerged/strain-trees-output/output_50_E_coli/align.phy -t /panfs/panfs1.ucsd.edu/panscratch/jhc103/VertMetaphlan-frmerged/strain-trees-output/output_50_E_coli/tree_parsimony_tips_changed.nwk -n GTRGAMMA_dist -w /panfs/panfs1.ucsd.edu/panscratch/jhc103/VertMetaphlan-frmerged/strain-trees-output/output_50_E_coli \n",
      "\n",
      "\n",
      "\n",
      "Log Likelihood Score after parameter optimization: -32980.344014\n",
      "\n",
      "\n",
      "Computing pairwise ML-distances ...\n",
      "\n",
      "Time for pair-wise ML distance computation of 4278 distances: 749.987782 seconds\n",
      "\n",
      "Distances written to file: /panfs/panfs1.ucsd.edu/panscratch/jhc103/VertMetaphlan-frmerged/strain-trees-output/output_50_E_coli/RAxML_distances.GTRGAMMA_dist\n"
     ]
    }
   ],
   "source": [
    "!/home/jhc103/miniconda3/envs/mpa3-py37/bin/raxmlHPC-PTHREADS-SSE3 -T 16 -f x -m GTRGAMMA -s $output_dir/align.phy -t $output_dir/tree_parsimony_tips_changed.nwk -n GTRGAMMA_dist -w $output_dir"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 46,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "['',\n",
       " '',\n",
       " 'This is RAxML version 8.2.12 released by Alexandros Stamatakis on May 2018.',\n",
       " '',\n",
       " 'With greatly appreciated code contributions by:',\n",
       " 'Andre Aberer      (HITS)',\n",
       " 'Simon Berger      (HITS)',\n",
       " 'Alexey Kozlov     (HITS)',\n",
       " 'Kassian Kobert    (HITS)',\n",
       " 'David Dao         (KIT and HITS)',\n",
       " 'Sarah Lutteropp   (KIT and HITS)',\n",
       " 'Nick Pattengale   (Sandia)',\n",
       " 'Wayne Pfeiffer    (SDSC)',\n",
       " 'Akifumi S. Tanabe (NRIFS)',\n",
       " 'Charlie Taylor    (UF)',\n",
       " '',\n",
       " '',\n",
       " 'Please also consult the RAxML-manual',\n",
       " '',\n",
       " 'Please report bugs via the RAxML google group!',\n",
       " 'Please send us all input files, the exact invocation, details of the HW and operating system,',\n",
       " 'as well as all error messages printed to screen.',\n",
       " '',\n",
       " '',\n",
       " 'raxmlHPC[-SSE3|-AVX|-PTHREADS|-PTHREADS-SSE3|-PTHREADS-AVX|-HYBRID|-HYBRID-SSE3|HYBRID-AVX]',\n",
       " '      -s sequenceFileName -n outputFileName -m substitutionModel',\n",
       " '      [-a weightFileName] [-A secondaryStructureSubstModel]',\n",
       " '      [-b bootstrapRandomNumberSeed] [-B wcCriterionThreshold]',\n",
       " '      [-c numberOfCategories] [-C] [-d] [-D]',\n",
       " '      [-e likelihoodEpsilon] [-E excludeFileName]',\n",
       " '      [-f a|A|b|B|c|C|d|D|e|E|F|g|G|h|H|i|I|j|J|k|m|n|N|o|p|P|q|r|R|s|S|t|T|u|v|V|w|W|x|y] [-F]',\n",
       " '      [-g groupingFileName] [-G placementThreshold] [-h] [-H]',\n",
       " '      [-i initialRearrangementSetting] [-I autoFC|autoMR|autoMRE|autoMRE_IGN]',\n",
       " '      [-j] [-J MR|MR_DROP|MRE|STRICT|STRICT_DROP|T_<PERCENT>] [-k] [-K] ',\n",
       " '      [-L MR|MRE|T_<PERCENT>] [-M]',\n",
       " '      [-o outGroupName1[,outGroupName2[,...]]][-O]',\n",
       " '      [-p parsimonyRandomSeed] [-P proteinModel]',\n",
       " '      [-q multipleModelFileName] [-r binaryConstraintTree]',\n",
       " '      [-R binaryModelParamFile] [-S secondaryStructureFile] [-t userStartingTree]',\n",
       " '      [-T numberOfThreads] [-u] [-U] [-v] [-V] [-w outputDirectory] [-W slidingWindowSize]',\n",
       " '      [-x rapidBootstrapRandomNumberSeed] [-X] [-y] [-Y quartetGroupingFileName|ancestralSequenceCandidatesFileName]',\n",
       " '      [-z multipleTreesFile] [-#|-N numberOfRuns|autoFC|autoMR|autoMRE|autoMRE_IGN]',\n",
       " '      [--mesquite][--silent][--no-seq-check][--no-bfgs]',\n",
       " '      [--asc-corr=stamatakis|felsenstein|lewis]',\n",
       " '      [--flag-check][--auto-prot=ml|bic|aic|aicc]',\n",
       " '      [--epa-keep-placements=number][--epa-accumulated-threshold=threshold]',\n",
       " '      [--epa-prob-threshold=threshold]',\n",
       " '      [--JC69][--K80][--HKY85]',\n",
       " '      [--bootstop-perms=number]',\n",
       " '',\n",
       " '      [--quartets-without-replacement]',\n",
       " '',\n",
       " '      [---without-replacement]',\n",
       " '',\n",
       " '      [--print-identical-sequences]',\n",
       " '',\n",
       " '      -a      Specify a column weight file name to assign individual weights to each column of ',\n",
       " '              the alignment. Those weights must be integers separated by any type and number ',\n",
       " '              of whitespaces whithin a separate file, see file \"example_weights\" for an example.',\n",
       " '',\n",
       " '      -A      Specify one of the secondary structure substitution models implemented in RAxML.',\n",
       " '              The same nomenclature as in the PHASE manual is used, available models: ',\n",
       " '              S6A, S6B, S6C, S6D, S6E, S7A, S7B, S7C, S7D, S7E, S7F, S16, S16A, S16B',\n",
       " '',\n",
       " '              DEFAULT: 16-state GTR model (S16)',\n",
       " '',\n",
       " '      -b      Specify an integer number (random seed) and turn on bootstrapping',\n",
       " '',\n",
       " '              DEFAULT: OFF',\n",
       " '',\n",
       " '      -B      specify a floating point number between 0.0 and 1.0 that will be used as cutoff threshold ',\n",
       " '              for the MR-based bootstopping criteria. The recommended setting is 0.03.',\n",
       " '',\n",
       " '              DEFAULT: 0.03 (recommended empirically determined setting)',\n",
       " '',\n",
       " '      -c      Specify number of distinct rate catgories for RAxML when model of rate heterogeneity',\n",
       " '              is set to CAT',\n",
       " '              Individual per-site rates are categorized into numberOfCategories rate ',\n",
       " '              categories to accelerate computations. ',\n",
       " '',\n",
       " '              DEFAULT: 25',\n",
       " '',\n",
       " '      -C      Enable verbose output for the \"-L\" and \"-f i\" options. This will produce more, as well as more verbose output files',\n",
       " '',\n",
       " '              DEFAULT: OFF',\n",
       " '',\n",
       " '      -d      start ML optimization from random starting tree ',\n",
       " '',\n",
       " '              DEFAULT: OFF',\n",
       " '',\n",
       " '      -D      ML search convergence criterion. This will break off ML searches if the relative ',\n",
       " '              Robinson-Foulds distance between the trees obtained from two consecutive lazy SPR cycles',\n",
       " '              is smaller or equal to 1%. Usage recommended for very large datasets in terms of taxa.',\n",
       " '              On trees with more than 500 taxa this will yield execution time improvements of approximately 50%',\n",
       " '              While yielding only slightly worse trees.',\n",
       " '',\n",
       " '              DEFAULT: OFF',\n",
       " '',\n",
       " '      -e      set model optimization precision in log likelihood units for final',\n",
       " '              optimization of tree topology',\n",
       " '',\n",
       " '              DEFAULT: 0.1   for models not using proportion of invariant sites estimate',\n",
       " '                       0.001 for models using proportion of invariant sites estimate',\n",
       " '',\n",
       " '      -E      specify an exclude file name, that contains a specification of alignment positions you wish to exclude.',\n",
       " '              Format is similar to Nexus, the file shall contain entries like \"100-200 300-400\", to exclude a',\n",
       " '              single column write, e.g., \"100-100\", if you use a mixed model, an appropriately adapted model file',\n",
       " '              will be written.',\n",
       " '',\n",
       " '      -f      select algorithm:',\n",
       " '',\n",
       " '              \"-f a\": rapid Bootstrap analysis and search for best-scoring ML tree in one program run',\n",
       " '              \"-f A\": compute marginal ancestral states on a ROOTED reference tree provided with \"-t\"',\n",
       " '              \"-f b\": draw bipartition information on a tree provided with \"-t\" based on multiple trees',\n",
       " '                      (e.g., from a bootstrap) in a file specified by \"-z\"',\n",
       " '              \"-f B\": optimize br-len scaler and other model parameters (GTR, alpha, etc.) on a tree provided with \"-t\".',\n",
       " '                      The tree needs to contain branch lengths. The branch lengths will not be optimized, just scaled by a single common value.',\n",
       " '              \"-f c\": check if the alignment can be properly read by RAxML',\n",
       " '              \"-f C\": ancestral sequence test for Jiajie, users will also need to provide a list of taxon names via -Y separated by whitespaces',\n",
       " '              \"-f d\": new rapid hill-climbing ',\n",
       " '                      DEFAULT: ON',\n",
       " '              \"-f D\": rapid hill-climbing with RELL bootstraps',\n",
       " '              \"-f e\": optimize model+branch lengths for given input tree under GAMMA/GAMMAI only',\n",
       " '              \"-f E\": execute very fast experimental tree search, at present only for testing',\n",
       " '              \"-f F\": execute fast experimental tree search, at present only for testing',\n",
       " '              \"-f g\": compute per site log Likelihoods for one or more trees passed via',\n",
       " '                      \"-z\" and write them to a file that can be read by CONSEL',\n",
       " '                      The model parameters will be estimated on the first tree only!',\n",
       " '              \"-f G\": compute per site log Likelihoods for one or more trees passed via',\n",
       " '                      \"-z\" and write them to a file that can be read by CONSEL.',\n",
       " '                      The model parameters will be re-estimated for each tree',\n",
       " '              \"-f h\": compute log likelihood test (SH-test) between best tree passed via \"-t\"',\n",
       " '                      and a bunch of other trees passed via \"-z\" ',\n",
       " '                      The model parameters will be estimated on the first tree only!',\n",
       " '              \"-f H\": compute log likelihood test (SH-test) between best tree passed via \"-t\"',\n",
       " '                      and a bunch of other trees passed via \"-z\" ',\n",
       " '                      The model parameters will be re-estimated for each tree',\n",
       " '              \"-f i\": calculate IC and TC scores (Salichos and Rokas 2013) on a tree provided with \"-t\" based on multiple trees',\n",
       " '                      (e.g., from a bootstrap) in a file specified by \"-z\"',\n",
       " '              \"-f I\": a simple tree rooting algorithm for unrooted trees.',\n",
       " '                      It roots the tree by rooting it at the branch that best balances the subtree lengths',\n",
       " '                      (sum over branches in the subtrees) of the left and right subtree.',\n",
       " '                      A branch with an optimal balance does not always exist!',\n",
       " '                      You need to specify the tree you want to root via \"-t\".',\n",
       " '              \"-f j\": generate a bunch of bootstrapped alignment files from an original alignemnt file.',\n",
       " '                      You need to specify a seed with \"-b\" and the number of replicates with \"-#\" ',\n",
       " '              \"-f J\": Compute SH-like support values on a given tree passed via \"-t\".',\n",
       " '              \"-f k\": Fix long branch lengths in partitioned data sets with missing data using the',\n",
       " '                      branch length stealing algorithm.',\n",
       " '                      This option only works in conjunction with \"-t\", \"-M\", and \"-q\".',\n",
       " '                      It will print out a tree with shorter branch lengths, but having the same likelihood score.',\n",
       " '              \"-f m\": compare bipartitions between two bunches of trees passed via \"-t\" and \"-z\" ',\n",
       " '                      respectively. This will return the Pearson correlation between all bipartitions found',\n",
       " '                      in the two tree files. A file called RAxML_bipartitionFrequencies.outpuFileName',\n",
       " '                      will be printed that contains the pair-wise bipartition frequencies of the two sets',\n",
       " '              \"-f n\": compute the log likelihood score of all trees contained in a tree file provided by',\n",
       " '                      \"-z\" under GAMMA or GAMMA+P-Invar',\n",
       " '                      The model parameters will be estimated on the first tree only!',\n",
       " '              \"-f N\": compute the log likelihood score of all trees contained in a tree file provided by',\n",
       " '                      \"-z\" under GAMMA or GAMMA+P-Invar',\n",
       " '                      The model parameters will be re-estimated for each tree',\n",
       " '              \"-f o\": old and slower rapid hill-climbing without heuristic cutoff',\n",
       " '              \"-f p\": perform pure stepwise MP addition of new sequences to an incomplete starting tree and exit',\n",
       " '              \"-f P\": perform a phylogenetic placement of sub trees specified in a file passed via \"-z\" into a given reference tree',\n",
       " '                      in which these subtrees are contained that is passed via \"-t\" using the evolutionary placement algorithm.',\n",
       " '              \"-f q\": fast quartet calculator',\n",
       " '              \"-f r\": compute pairwise Robinson-Foulds (RF) distances between all pairs of trees in a tree file passed via \"-z\" ',\n",
       " '                      if the trees have node labales represented as integer support values the program will also compute two flavors of',\n",
       " '                      the weighted Robinson-Foulds (WRF) distance',\n",
       " '              \"-f R\": compute all pairwise Robinson-Foulds (RF) distances between a large reference tree  passed via \"-t\" ',\n",
       " '                      and many smaller trees (that must have a subset of the taxa of the large tree) passed via \"-z\".',\n",
       " '                      This option is intended for checking the plausibility of very large phylogenies that can not be inspected',\n",
       " '                      visually any more.',\n",
       " '              \"-f s\": split up a multi-gene partitioned alignment into the respective subalignments ',\n",
       " '              \"-f S\": compute site-specific placement bias using a leave one out test inspired by the evolutionary placement algorithm',\n",
       " '              \"-f t\": do randomized tree searches on one fixed starting tree',\n",
       " '              \"-f T\": do final thorough optimization of ML tree from rapid bootstrap search in stand-alone mode',\n",
       " '              \"-f u\": execute morphological weight calibration using maximum likelihood, this will return a weight vector.',\n",
       " '                      you need to provide a morphological alignment and a reference tree via \"-t\" ',\n",
       " '              \"-f v\": classify a bunch of environmental sequences into a reference tree using thorough read insertions',\n",
       " '                      you will need to start RAxML with a non-comprehensive reference tree and an alignment containing all sequences (reference + query)',\n",
       " '              \"-f V\": classify a bunch of environmental sequences into a reference tree using thorough read insertions',\n",
       " '                      you will need to start RAxML with a non-comprehensive reference tree and an alignment containing all sequences (reference + query)',\n",
       " '                      WARNING: this is a test implementation for more efficient handling of multi-gene/whole-genome datasets!',\n",
       " '              \"-f w\": compute ELW test on a bunch of trees passed via \"-z\" ',\n",
       " '                      The model parameters will be estimated on the first tree only!',\n",
       " '              \"-f W\": compute ELW test on a bunch of trees passed via \"-z\" ',\n",
       " '                      The model parameters will be re-estimated for each tree',\n",
       " '              \"-f x\": compute pair-wise ML distances, ML model parameters will be estimated on an MP ',\n",
       " '                      starting tree or a user-defined tree passed via \"-t\", only allowed for GAMMA-based',\n",
       " '                      models of rate heterogeneity',\n",
       " '              \"-f y\": classify a bunch of environmental sequences into a reference tree using parsimony',\n",
       " '                      you will need to start RAxML with a non-comprehensive reference tree and an alignment containing all sequences (reference + query)',\n",
       " '',\n",
       " '              DEFAULT for \"-f\": new rapid hill climbing',\n",
       " '',\n",
       " '',\n",
       " '      -F      enable ML tree searches under CAT model for very large trees without switching to ',\n",
       " '              GAMMA in the end (saves memory).',\n",
       " '              This option can also be used with the GAMMA models in order to avoid the thorough optimization ',\n",
       " '              of the best-scoring ML tree in the end.',\n",
       " '',\n",
       " '              DEFAULT: OFF',\n",
       " '',\n",
       " '      -g      specify the file name of a multifurcating constraint tree',\n",
       " '              this tree does not need to be comprehensive, i.e. must not contain all taxa',\n",
       " '',\n",
       " '      -G      enable the ML-based evolutionary placement algorithm heuristics',\n",
       " '              by specifiyng a threshold value (fraction of insertion branches to be evaluated',\n",
       " '              using slow insertions under ML).',\n",
       " '',\n",
       " '      -h      Display this help message.',\n",
       " '',\n",
       " '      -H      Disable pattern compression.',\n",
       " '',\n",
       " '              DEFAULT: ON',\n",
       " '',\n",
       " '      -i      Initial rearrangement setting for the subsequent application of topological ',\n",
       " '              changes phase',\n",
       " '',\n",
       " '      -I      a posteriori bootstopping analysis. Use:',\n",
       " '             \"-I autoFC\" for the frequency-based criterion',\n",
       " '             \"-I autoMR\" for the majority-rule consensus tree criterion',\n",
       " '             \"-I autoMRE\" for the extended majority-rule consensus tree criterion',\n",
       " '             \"-I autoMRE_IGN\" for metrics similar to MRE, but include bipartitions under the threshold whether they are compatible',\n",
       " '                              or not. This emulates MRE but is faster to compute.',\n",
       " '              You also need to pass a tree file containg several bootstrap replicates via \"-z\" ',\n",
       " '',\n",
       " '      -j      Specifies that intermediate tree files shall be written to file during the standard ML and BS tree searches.',\n",
       " '',\n",
       " '              DEFAULT: OFF',\n",
       " '',\n",
       " '      -J      Compute majority rule consensus tree with \"-J MR\" or extended majority rule consensus tree with \"-J MRE\"',\n",
       " '              or strict consensus tree with \"-J STRICT\". For a custom consensus threshold >= 50%, specify T_<NUM>, where 100 >= NUM >= 50.',\n",
       " '              Options \"-J STRICT_DROP\" and \"-J MR_DROP\" will execute an algorithm that identifies dropsets which contain',\n",
       " '              rogue taxa as proposed by Pattengale et al. in the paper \"Uncovering hidden phylogenetic consensus\".',\n",
       " '              You will also need to provide a tree file containing several UNROOTED trees via \"-z\"',\n",
       " '',\n",
       " '      -k      Specifies that bootstrapped trees should be printed with branch lengths.',\n",
       " '              The bootstraps will run a bit longer, because model parameters will be optimized',\n",
       " '              at the end of each run under GAMMA or GAMMA+P-Invar respectively.',\n",
       " '',\n",
       " '              DEFAULT: OFF',\n",
       " '',\n",
       " '      -K      Specify one of the multi-state substitution models (max 32 states) implemented in RAxML.',\n",
       " '              Available models are: ORDERED, MK, GTR',\n",
       " '',\n",
       " '              DEFAULT: GTR model ',\n",
       " '',\n",
       " '      -L     Compute consensus trees labelled by IC supports and the overall TC value as proposed in Salichos and Rokas 2013.',\n",
       " '             Compute a majority rule consensus tree with \"-L MR\" or an extended majority rule consensus tree with \"-L MRE\".',\n",
       " '             For a custom consensus threshold >= 50%, specify \"-L T_<NUM>\", where 100 >= NUM >= 50.',\n",
       " '             You will of course also need to provide a tree file containing several UNROOTED trees via \"-z\"!',\n",
       " '',\n",
       " '      -m      Model of Binary (Morphological), Nucleotide, Multi-State, or Amino Acid Substitution: ',\n",
       " '',\n",
       " '              BINARY:',\n",
       " '',\n",
       " '                \"-m BINCAT[X]\"       : Optimization of site-specific',\n",
       " '                                       evolutionary rates which are categorized into numberOfCategories distinct ',\n",
       " '                                       rate categories for greater computational efficiency. Final tree might be evaluated',\n",
       " '                                       automatically under BINGAMMA, depending on the tree search option.',\n",
       " '                                       With the optional \"X\" appendix you can specify a ML estimate of base frequencies.',\n",
       " '                \"-m BINCATI[X]\"      : Optimization of site-specific',\n",
       " '                                       evolutionary rates which are categorized into numberOfCategories distinct ',\n",
       " '                                       rate categories for greater computational efficiency. Final tree might be evaluated',\n",
       " '                                       automatically under BINGAMMAI, depending on the tree search option.',\n",
       " '                                       With the optional \"X\" appendix you can specify a ML estimate of base frequencies.',\n",
       " '                \"-m ASC_BINCAT[X]\"   : Optimization of site-specific',\n",
       " '                                       evolutionary rates which are categorized into numberOfCategories distinct ',\n",
       " '                                       rate categories for greater computational efficiency. Final tree might be evaluated',\n",
       " '                                       automatically under BINGAMMA, depending on the tree search option.',\n",
       " '                                       With the optional \"X\" appendix you can specify a ML estimate of base frequencies.',\n",
       " '                                       The ASC prefix willl correct the likelihood for ascertainment bias.',\n",
       " '                \"-m BINGAMMA[X]\"     : GAMMA model of rate heterogeneity (alpha parameter will be estimated).',\n",
       " '                                       With the optional \"X\" appendix you can specify a ML estimate of base frequencies.',\n",
       " '                \"-m ASC_BINGAMMA[X]\" : GAMMA model of rate heterogeneity (alpha parameter will be estimated).',\n",
       " '                                       The ASC prefix willl correct the likelihood for ascertainment bias.',\n",
       " '                                       With the optional \"X\" appendix you can specify a ML estimate of base frequencies.',\n",
       " '                \"-m BINGAMMAI[X]\"    : Same as BINGAMMA, but with estimate of proportion of invariable sites.',\n",
       " '                                       With the optional \"X\" appendix you can specify a ML estimate of base frequencies.',\n",
       " '',\n",
       " '              NUCLEOTIDES:',\n",
       " '',\n",
       " '                \"-m GTRCAT[X]\"       : GTR + Optimization of substitution rates + Optimization of site-specific',\n",
       " '                                       evolutionary rates which are categorized into numberOfCategories distinct ',\n",
       " '                                       rate categories for greater computational efficiency.  Final tree might be evaluated',\n",
       " '                                       under GTRGAMMA, depending on the tree search option.',\n",
       " '                                       With the optional \"X\" appendix you can specify a ML estimate of base frequencies.',\n",
       " '                \"-m GTRCATI[X]\"      : GTR + Optimization of substitution rates + Optimization of site-specific',\n",
       " '                                       evolutionary rates which are categorized into numberOfCategories distinct ',\n",
       " '                                       rate categories for greater computational efficiency.  Final tree might be evaluated',\n",
       " '                                       under GTRGAMMAI, depending on the tree search option.',\n",
       " '                                       With the optional \"X\" appendix you can specify a ML estimate of base frequencies.',\n",
       " '                \"-m ASC_GTRCAT[X]\"   : GTR + Optimization of substitution rates + Optimization of site-specific',\n",
       " '                                       evolutionary rates which are categorized into numberOfCategories distinct ',\n",
       " '                                       rate categories for greater computational efficiency.  Final tree might be evaluated',\n",
       " '                                       under GTRGAMMA, depending on the tree search option.',\n",
       " '                                       With the optional \"X\" appendix you can specify a ML estimate of base frequencies.',\n",
       " '                                       The ASC prefix willl correct the likelihood for ascertainment bias.',\n",
       " '                \"-m GTRGAMMA[X]\"     : GTR + Optimization of substitution rates + GAMMA model of rate ',\n",
       " '                                       heterogeneity (alpha parameter will be estimated).',\n",
       " '                                       With the optional \"X\" appendix you can specify a ML estimate of base frequencies.',\n",
       " '                \"-m ASC_GTRGAMMA[X]\" : GTR + Optimization of substitution rates + GAMMA model of rate ',\n",
       " '                                       heterogeneity (alpha parameter will be estimated).',\n",
       " '                                       The ASC prefix willl correct the likelihood for ascertainment bias.',\n",
       " '                                       With the optional \"X\" appendix you can specify a ML estimate of base frequencies.',\n",
       " '                \"-m GTRGAMMAI[X]\"    : Same as GTRGAMMA, but with estimate of proportion of invariable sites.',\n",
       " '                                       With the optional \"X\" appendix you can specify a ML estimate of base frequencies.',\n",
       " '',\n",
       " '              MULTI-STATE:',\n",
       " '',\n",
       " '                \"-m MULTICAT[X]\"       : Optimization of site-specific',\n",
       " '                                         evolutionary rates which are categorized into numberOfCategories distinct ',\n",
       " '                                         rate categories for greater computational efficiency. Final tree might be evaluated',\n",
       " '                                         automatically under MULTIGAMMA, depending on the tree search option.',\n",
       " '                                         With the optional \"X\" appendix you can specify a ML estimate of base frequencies.',\n",
       " '                \"-m MULTICATI[X]\"      : Optimization of site-specific',\n",
       " '                                         evolutionary rates which are categorized into numberOfCategories distinct ',\n",
       " '                                         rate categories for greater computational efficiency. Final tree might be evaluated',\n",
       " '                                         automatically under MULTIGAMMAI, depending on the tree search option.',\n",
       " '                                         With the optional \"X\" appendix you can specify a ML estimate of base frequencies.',\n",
       " '                \"-m ASC_MULTICAT[X]\"   : Optimization of site-specific',\n",
       " '                                         evolutionary rates which are categorized into numberOfCategories distinct ',\n",
       " '                                         rate categories for greater computational efficiency. Final tree might be evaluated',\n",
       " '                                         automatically under MULTIGAMMA, depending on the tree search option.',\n",
       " '                                         With the optional \"X\" appendix you can specify a ML estimate of base frequencies.',\n",
       " '                                         The ASC prefix willl correct the likelihood for ascertainment bias.',\n",
       " '                \"-m MULTIGAMMA[X]\"     : GAMMA model of rate heterogeneity (alpha parameter will be estimated).',\n",
       " '                                         With the optional \"X\" appendix you can specify a ML estimate of base frequencies.',\n",
       " '                \"-m ASC_MULTIGAMMA[X]\" : GAMMA model of rate heterogeneity (alpha parameter will be estimated).',\n",
       " '                                         The ASC prefix willl correct the likelihood for ascertainment bias.',\n",
       " '                                         With the optional \"X\" appendix you can specify a ML estimate of base frequencies.',\n",
       " '                \"-m MULTIGAMMAI[X]\"    : Same as MULTIGAMMA, but with estimate of proportion of invariable sites.',\n",
       " '                                         With the optional \"X\" appendix you can specify a ML estimate of base frequencies.',\n",
       " '',\n",
       " '                You can use up to 32 distinct character states to encode multi-state regions, they must be used in the following order:',\n",
       " '                0, 1, 2, 3, 4, 5, 6, 7, 8, 9, A, B, C, D, E, F, G, H, I, J, K, L, M, N, O, P, Q, R, S, T, U, V',\n",
       " '                i.e., if you have 6 distinct character states you would use 0, 1, 2, 3, 4, 5 to encode these.',\n",
       " '                The substitution model for the multi-state regions can be selected via the \"-K\" option',\n",
       " '',\n",
       " '              AMINO ACIDS:',\n",
       " '',\n",
       " '                \"-m PROTCATmatrixName[F|X]\"       : specified AA matrix + Optimization of substitution rates + Optimization of site-specific',\n",
       " '                                                    evolutionary rates which are categorized into numberOfCategories distinct ',\n",
       " '                                                    rate categories for greater computational efficiency.   Final tree might be evaluated',\n",
       " '                                                    automatically under PROTGAMMAmatrixName[F|X], depending on the tree search option.',\n",
       " '                                                    With the optional \"X\" appendix you can specify a ML estimate of base frequencies.',\n",
       " '                \"-m PROTCATImatrixName[F|X]\"      : specified AA matrix + Optimization of substitution rates + Optimization of site-specific',\n",
       " '                                                    evolutionary rates which are categorized into numberOfCategories distinct ',\n",
       " '                                                    rate categories for greater computational efficiency.   Final tree might be evaluated',\n",
       " '                                                    automatically under PROTGAMMAImatrixName[F|X], depending on the tree search option.',\n",
       " '                                                    With the optional \"X\" appendix you can specify a ML estimate of base frequencies.',\n",
       " '                \"-m ASC_PROTCATmatrixName[F|X]\"   : specified AA matrix + Optimization of substitution rates + Optimization of site-specific',\n",
       " '                                                    evolutionary rates which are categorized into numberOfCategories distinct ',\n",
       " '                                                    rate categories for greater computational efficiency.   Final tree might be evaluated',\n",
       " '                                                    automatically under PROTGAMMAmatrixName[F|X], depending on the tree search option.',\n",
       " '                                                    With the optional \"X\" appendix you can specify a ML estimate of base frequencies.',\n",
       " '                                                    The ASC prefix willl correct the likelihood for ascertainment bias.',\n",
       " '                \"-m PROTGAMMAmatrixName[F|X]\"     : specified AA matrix + Optimization of substitution rates + GAMMA model of rate ',\n",
       " '                                                    heterogeneity (alpha parameter will be estimated).',\n",
       " '                                                    With the optional \"X\" appendix you can specify a ML estimate of base frequencies.',\n",
       " '                \"-m ASC_PROTGAMMAmatrixName[F|X]\" : specified AA matrix + Optimization of substitution rates + GAMMA model of rate ',\n",
       " '                                                    heterogeneity (alpha parameter will be estimated).',\n",
       " '                                                    The ASC prefix willl correct the likelihood for ascertainment bias.',\n",
       " '                                                    With the optional \"X\" appendix you can specify a ML estimate of base frequencies.',\n",
       " '                \"-m PROTGAMMAImatrixName[F|X]\"    : Same as PROTGAMMAmatrixName[F|X], but with estimate of proportion of invariable sites.',\n",
       " '                                                    With the optional \"X\" appendix you can specify a ML estimate of base frequencies.',\n",
       " '',\n",
       " '                Available AA substitution models:',\n",
       " '                DAYHOFF, DCMUT, JTT, MTREV, WAG, RTREV, CPREV, VT, ',\n",
       " '                BLOSUM62, MTMAM, LG, MTART, MTZOA, PMB, HIVB, HIVW, ',\n",
       " '                JTTDCMUT, FLU, STMTREV, DUMMY, DUMMY2, AUTO, LG4M, LG4X, ',\n",
       " '                PROT_FILE, GTR_UNLINKED, GTR',\n",
       " '                With the optional \"F\" appendix you can specify if you want to use empirical base frequencies.',\n",
       " '                AUTOF and AUTOX are not supported any more, if you specify AUTO it will test prot subst. models with and without empirical',\n",
       " '                base frequencies now!',\n",
       " '                Please note that for partitioned models you can in addition specify the per-gene AA model in',\n",
       " '                the partition file (see manual for details). Also note that if you estimate AA GTR parameters on a partitioned',\n",
       " '                dataset, they will be linked (estimated jointly) across all partitions to avoid over-parametrization',\n",
       " '',\n",
       " '      -M      Switch on estimation of individual per-partition branch lengths. Only has effect when used in combination with \"-q\"',\n",
       " '              Branch lengths for individual partitions will be printed to separate files',\n",
       " '              A weighted average of the branch lengths is computed by using the respective partition lengths',\n",
       " '',\n",
       " '              DEFAULT: OFF',\n",
       " '',\n",
       " '      -n      Specifies the name of the output file.',\n",
       " '',\n",
       " '      -o      Specify the name of a single outgroup or a comma-separated list of outgroups, eg \"-o Rat\" ',\n",
       " '              or \"-o Rat,Mouse\", in case that multiple outgroups are not monophyletic the first name ',\n",
       " \"              in the list will be selected as outgroup, don't leave spaces between taxon names!\",\n",
       " '',\n",
       " '      -O      Disable check for completely undetermined sequence in alignment.',\n",
       " '              The program will not exit with an error message when \"-O\" is specified.',\n",
       " '',\n",
       " '              DEFAULT: check enabled',\n",
       " '',\n",
       " '      -p      Specify a random number seed for the parsimony inferences. This allows you to reproduce your results',\n",
       " '              and will help me debug the program.',\n",
       " '',\n",
       " '      -P      Specify the file name of a user-defined AA (Protein) substitution model. This file must contain',\n",
       " '              420 entries, the first 400 being the AA substitution rates (this must be a symmetric matrix) and the',\n",
       " '              last 20 are the empirical base frequencies',\n",
       " '',\n",
       " '      -q      Specify the file name which contains the assignment of models to alignment',\n",
       " '              partitions for multiple models of substitution. For the syntax of this file',\n",
       " '              please consult the manual.',\n",
       " '',\n",
       " '      -r      Specify the file name of a binary constraint tree.',\n",
       " '              this tree does not need to be comprehensive, i.e. must not contain all taxa',\n",
       " '',\n",
       " '      -R      Specify the file name of a binary model parameter file that has previously been generated',\n",
       " '              with RAxML using the -f e tree evaluation option. The file name should be: ',\n",
       " '              RAxML_binaryModelParameters.runID',\n",
       " '',\n",
       " '      -s      Specify the name of the alignment data file in PHYLIP format',\n",
       " '',\n",
       " '      -S      Specify the name of a secondary structure file. The file can contain \".\" for ',\n",
       " '              alignment columns that do not form part of a stem and characters \"()<>[]{}\" to define ',\n",
       " '              stem regions and pseudoknots',\n",
       " '',\n",
       " '      -t      Specify a user starting tree file name in Newick format',\n",
       " '',\n",
       " '      -T      PTHREADS VERSION ONLY! Specify the number of threads you want to run.',\n",
       " '              Make sure to set \"-T\" to at most the number of CPUs you have on your machine,',\n",
       " '              otherwise, there will be a huge performance decrease!',\n",
       " '',\n",
       " '      -u      use the median for the discrete approximation of the GAMMA model of rate heterogeneity',\n",
       " '',\n",
       " '              DEFAULT: OFF',\n",
       " '',\n",
       " '      -U      Try to save memory by using SEV-based implementation for gap columns on large gappy alignments',\n",
       " '              The technique is described here: http://www.biomedcentral.com/1471-2105/12/470',\n",
       " '              This will only work for DNA and/or PROTEIN data and only with the SSE3 or AVX-vextorized version of the code.',\n",
       " '',\n",
       " '      -v      Display version information',\n",
       " '',\n",
       " '      -V      Disable rate heterogeneity among sites model and use one without rate heterogeneity instead.',\n",
       " '              Only works if you specify the CAT model of rate heterogeneity.',\n",
       " '',\n",
       " '              DEFAULT: use rate heterogeneity',\n",
       " '',\n",
       " '      -w      FULL (!) path to the directory into which RAxML shall write its output files',\n",
       " '',\n",
       " '              DEFAULT: current directory',\n",
       " '',\n",
       " '      -W      Sliding window size for leave-one-out site-specific placement bias algorithm',\n",
       " '              only effective when used in combination with \"-f S\" ',\n",
       " '',\n",
       " '              DEFAULT: 100 sites',\n",
       " '',\n",
       " '      -x      Specify an integer number (random seed) and turn on rapid bootstrapping',\n",
       " '              CAUTION: unlike in version 7.0.4 RAxML will conduct rapid BS replicates under ',\n",
       " '              the model of rate heterogeneity you specified via \"-m\" and not by default under CAT',\n",
       " '',\n",
       " '      -X      Same as the \"-y\" option below, however the parsimony search is more superficial.',\n",
       " '              RAxML will only do a randomized stepwise addition order parsimony tree reconstruction',\n",
       " '              without performing any additional SPRs.',\n",
       " '              This may be helpful for very broad whole-genome datasets, since this can generate topologically',\n",
       " '              more different starting trees.',\n",
       " '',\n",
       " '              DEFAULT: OFF',\n",
       " '',\n",
       " '      -y      If you want to only compute a parsimony starting tree with RAxML specify \"-y\",',\n",
       " '              the program will exit after computation of the starting tree',\n",
       " '',\n",
       " '              DEFAULT: OFF',\n",
       " '',\n",
       " '      -Y      Pass a quartet grouping file name defining four groups from which to draw quartets',\n",
       " '              The file input format must contain 4 groups in the following form:',\n",
       " '              (Chicken, Human, Loach), (Cow, Carp), (Mouse, Rat, Seal), (Whale, Frog);',\n",
       " '              Only works in combination with -f q !',\n",
       " '',\n",
       " '      -z      Specify the file name of a file containing multiple trees e.g. from a bootstrap',\n",
       " '              that shall be used to draw bipartition values onto a tree provided with \"-t\",',\n",
       " '              It can also be used to compute per site log likelihoods in combination with \"-f g\"',\n",
       " '              and to read a bunch of trees for a couple of other options (\"-f h\", \"-f m\", \"-f n\").',\n",
       " '',\n",
       " '      -#|-N   Specify the number of alternative runs on distinct starting trees',\n",
       " '              In combination with the \"-b\" option, this will invoke a multiple boostrap analysis',\n",
       " '              Note that \"-N\" has been added as an alternative since \"-#\" sometimes caused problems',\n",
       " '              with certain MPI job submission systems, since \"-#\" is often used to start comments.',\n",
       " '              If you want to use the bootstopping criteria specify \"-# autoMR\" or \"-# autoMRE\" or \"-# autoMRE_IGN\"',\n",
       " '              for the majority-rule tree based criteria (see -I option) or \"-# autoFC\" for the frequency-based criterion.',\n",
       " '              Bootstopping will only work in combination with \"-x\" or \"-b\"',\n",
       " '',\n",
       " '              DEFAULT: 1 single analysis',\n",
       " '',\n",
       " '      --mesquite Print output files that can be parsed by Mesquite.',\n",
       " '',\n",
       " '              DEFAULT: Off',\n",
       " '',\n",
       " '      --silent Disables printout of warnings related to identical sequences and entirely undetermined sites in the alignment',\n",
       " '',\n",
       " '              DEFAULT: Off',\n",
       " '',\n",
       " '      --no-seq-check Disables checking the input MSA for identical sequences and entirely undetermined sites.',\n",
       " '                     Enabling this option may save time, in particular for large phylogenomic alignments.',\n",
       " '                     Before using this, make sure to check the alignment using the \"-f c\" option!',\n",
       " '',\n",
       " '              DEFAULT: Off ',\n",
       " '',\n",
       " '      --no-bfgs Disables automatic usage of BFGS method to optimize GTR rates on unpartitioned DNA datasets',\n",
       " '',\n",
       " '              DEFAULT: BFGS on',\n",
       " '',\n",
       " '      --asc-corr Allows to specify the type of ascertainment bias correction you wish to use. There are 3',\n",
       " '                 types available:',\n",
       " '                 --asc-corr=lewis: the standard correction by Paul Lewis',\n",
       " '                 --asc-corr=felsenstein: a correction introduced by Joe Felsenstein that allows to explicitely specify',\n",
       " '                                         the number of invariable sites (if known) one wants to correct for.',\n",
       " '                 --asc-corr=stamatakis: a correction introduced by myself that allows to explicitely specify',\n",
       " '                                        the number of invariable sites for each character (if known) one wants to correct for.',\n",
       " '',\n",
       " '      --flag-check When using this option, RAxML will only check if all command line flags specifed are available and then exit',\n",
       " '                   with a message listing all invalid command line flags or with a message stating that all flags are valid.',\n",
       " '',\n",
       " '      --auto-prot=ml|bic|aic|aicc When using automatic protein model selection you can chose the criterion for selecting these models.',\n",
       " '                  RAxML will test all available prot subst. models except for LG4M, LG4X and GTR-based models, with and without empirical base frequencies.',\n",
       " '                  You can chose between ML score based selection and the BIC, AIC, and AICc criteria.',\n",
       " '',\n",
       " '                  DEFAULT: ml',\n",
       " '',\n",
       " '      --epa-keep-placements=number specify the number of potential placements you want to keep for each read in the EPA algorithm.',\n",
       " '                  Note that, the actual values printed will also depend on the settings for --epa-prob-threshold=threshold !',\n",
       " '',\n",
       " '                  DEFAULT: 7',\n",
       " '',\n",
       " '      --epa-prob-threshold=threshold specify a percent threshold for including potential placements of a read depending on the ',\n",
       " '                  maximum placement weight for this read. If you set this value to 0.01 placements that have a placement weight of 1 per cent of',\n",
       " '                  the maximum placement will still be printed to file if the setting of --epa-keep-placements allows for it',\n",
       " '',\n",
       " '                  DEFAULT: 0.01',\n",
       " '',\n",
       " '      --epa-accumulated-threshold=threshold specify an accumulated likelihood weight threshold for which different placements of read are printed',\n",
       " '                  to file. Placements for a read will be printed until the sum of their placement weights has reached the threshold value.',\n",
       " '                  Note that, this option can neither be used in combination with --epa-prob-threshold nor with --epa-keep-placements!',\n",
       " '',\n",
       " '      --JC69 specify that all DNA partitions will evolve under the Jukes-Cantor model, this overrides all other model specifications for DNA partitions.',\n",
       " '',\n",
       " '                  DEFAULT: Off',\n",
       " '',\n",
       " '      --K80 specify that all DNA partitions will evolve under the K80 model, this overrides all other model specifications for DNA partitions.',\n",
       " '',\n",
       " '                  DEFAULT: Off',\n",
       " '',\n",
       " '      --HKY85 specify that all DNA partitions will evolve under the HKY85 model, this overrides all other model specifications for DNA partitions.',\n",
       " '',\n",
       " '                  DEFAULT: Off',\n",
       " '',\n",
       " '',\n",
       " '      --bootstop-perms=number specify the number of permutations to be conducted for the bootstopping/bootstrap convergence test.',\n",
       " '                  The allowed minimum number is 100!',\n",
       " '',\n",
       " '                  DEFAULT: 100',\n",
       " '',\n",
       " '      --quartets-without-replacement specify that quartets are randomly subsampled, but without replacement.',\n",
       " '',\n",
       " '                  DEFAULT: random sampling with replacements',\n",
       " '',\n",
       " '      --print-identical-sequences specify that RAxML shall automatically generate a .reduced alignment with all',\n",
       " '                  undetermined columns removed, but without removing exactly identical sequences',\n",
       " '',\n",
       " '                  DEFAULT: identical sequences will also be removed in the .reduced file',\n",
       " '',\n",
       " '',\n",
       " '',\n",
       " '',\n",
       " '']"
      ]
     },
     "execution_count": 46,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "!!/home/jhc103/miniconda3/envs/mpa3-py37/bin/raxmlHPC-PTHREADS-SSE3 -h"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "jupyter-general",
   "language": "python",
   "name": "jupyter-general"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.6.13"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
