{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [],
   "source": [
    "import pandas as pd\n",
    "from os.path import join\n",
    "import os"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {},
   "outputs": [],
   "source": [
    "## What marker_in_n_samples threshold you want to investigate?\n",
    "marker_threshold = \"50\" #65\n",
    "\n",
    "## Category of samples wishing to run! options are \"\", \"_mammallianHost\", \"_nonMammallianHost\"\n",
    "category = \"\"\n",
    "\n",
    "## Set up the directory_path to work in \n",
    "working_dir = \"/panfs/panfs1.ucsd.edu/panscratch/jhc103/VertMetaphlan-frmerged\"\n",
    "tree_dir = \"/panfs/panfs1.ucsd.edu/panscratch/jhc103/VertMetaphlan-frmerged/strain-trees-output\""
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {},
   "outputs": [],
   "source": [
    "## Import and adjust dataframe \n",
    "path_to_file = join(tree_dir, \"optimal_marker_per_sample_setting_for_\"+marker_threshold+\"%_marker_threshold\"+category+\".csv\")\n",
    "df_bac_marker=pd.read_csv(path_to_file,index_col=0)\n",
    "# df_bac_marker.drop(columns = ['Unnamed: 0'], inplace = True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>marker_threshold</th>\n",
       "      <th>number_of_samples</th>\n",
       "      <th>best_alignment_score</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>bacteria</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>s__Akkermansia_muciniphila.fna</th>\n",
       "      <td>17</td>\n",
       "      <td>21</td>\n",
       "      <td>1632</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>s__Bacteroides_dorei.fna</th>\n",
       "      <td>26</td>\n",
       "      <td>24</td>\n",
       "      <td>1854</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>s__Bacteroides_fragilis.fna</th>\n",
       "      <td>20</td>\n",
       "      <td>24</td>\n",
       "      <td>2255</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>s__Bacteroides_thetaiotaomicron.fna</th>\n",
       "      <td>29</td>\n",
       "      <td>16</td>\n",
       "      <td>1472</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>s__Bacteroides_uniformis.fna</th>\n",
       "      <td>13</td>\n",
       "      <td>40</td>\n",
       "      <td>3504</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>s__Bacteroides_vulgatus.fna</th>\n",
       "      <td>12</td>\n",
       "      <td>25</td>\n",
       "      <td>784</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>s__Clostridium_perfringens.fna</th>\n",
       "      <td>16</td>\n",
       "      <td>46</td>\n",
       "      <td>4787</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>s__Enterococcus_faecalis.fna</th>\n",
       "      <td>22</td>\n",
       "      <td>15</td>\n",
       "      <td>1526</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>s__Escherichia_coli.fna</th>\n",
       "      <td>6</td>\n",
       "      <td>92</td>\n",
       "      <td>2047</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>s__Lactobacillus_rodentium.fna</th>\n",
       "      <td>21</td>\n",
       "      <td>22</td>\n",
       "      <td>1047</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>s__Parabacteroides_distasonis.fna</th>\n",
       "      <td>42</td>\n",
       "      <td>18</td>\n",
       "      <td>1485</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>s__Prevotella_copri.fna</th>\n",
       "      <td>22</td>\n",
       "      <td>18</td>\n",
       "      <td>1912</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>s__Pseudomonas_yamanorum.fna</th>\n",
       "      <td>21</td>\n",
       "      <td>28</td>\n",
       "      <td>1023</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                     marker_threshold  number_of_samples  \\\n",
       "bacteria                                                                   \n",
       "s__Akkermansia_muciniphila.fna                     17                 21   \n",
       "s__Bacteroides_dorei.fna                           26                 24   \n",
       "s__Bacteroides_fragilis.fna                        20                 24   \n",
       "s__Bacteroides_thetaiotaomicron.fna                29                 16   \n",
       "s__Bacteroides_uniformis.fna                       13                 40   \n",
       "s__Bacteroides_vulgatus.fna                        12                 25   \n",
       "s__Clostridium_perfringens.fna                     16                 46   \n",
       "s__Enterococcus_faecalis.fna                       22                 15   \n",
       "s__Escherichia_coli.fna                             6                 92   \n",
       "s__Lactobacillus_rodentium.fna                     21                 22   \n",
       "s__Parabacteroides_distasonis.fna                  42                 18   \n",
       "s__Prevotella_copri.fna                            22                 18   \n",
       "s__Pseudomonas_yamanorum.fna                       21                 28   \n",
       "\n",
       "                                     best_alignment_score  \n",
       "bacteria                                                   \n",
       "s__Akkermansia_muciniphila.fna                       1632  \n",
       "s__Bacteroides_dorei.fna                             1854  \n",
       "s__Bacteroides_fragilis.fna                          2255  \n",
       "s__Bacteroides_thetaiotaomicron.fna                  1472  \n",
       "s__Bacteroides_uniformis.fna                         3504  \n",
       "s__Bacteroides_vulgatus.fna                           784  \n",
       "s__Clostridium_perfringens.fna                       4787  \n",
       "s__Enterococcus_faecalis.fna                         1526  \n",
       "s__Escherichia_coli.fna                              2047  \n",
       "s__Lactobacillus_rodentium.fna                       1047  \n",
       "s__Parabacteroides_distasonis.fna                    1485  \n",
       "s__Prevotella_copri.fna                              1912  \n",
       "s__Pseudomonas_yamanorum.fna                         1023  "
      ]
     },
     "execution_count": 10,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df_bac_marker"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {},
   "outputs": [],
   "source": [
    "run_strainphlan_original = False\n",
    "if run_strainphlan_original:\n",
    "    for index, row in df_bac_marker.iterrows():\n",
    "\n",
    "        ## Create the two bacteria name string you might want to utilize\n",
    "        bacteria_frontTrimmed = index.strip(\"s__\")\n",
    "        bacteria_backTrimmed = index.strip(\"fna\").strip(\".\")\n",
    "        bacteria_trimmed = bacteria_frontTrimmed.strip(\"fna\").strip(\".\")\n",
    "        bacteria_simplified=bacteria_trimmed[0]+bacteria_trimmed[bacteria_trimmed.find(\"_\"):len(bacteria_trimmed)]\n",
    "\n",
    "\n",
    "    #     ## marker_sample count\n",
    "        markers_per_sample = row['marker_threshold']\n",
    "\n",
    "        # Set up output directory then create it if not created yet \n",
    "        path_to_output_dir = join(tree_dir, \"output_\"+marker_threshold+\"_\"+bacteria_simplified)\n",
    "        if not os.path.isdir(path_to_output_dir):\n",
    "            os.mkdir(path_to_output_dir)\n",
    "\n",
    "        print(\"Running strainphlan on \"+ index +\":\")\n",
    "        print(path_to_output_dir)\n",
    "\n",
    "        ## Run strainphlan \n",
    "        !strainphlan -s $working_dir/consensus_markers$category/*.pkl -m $working_dir/db_markers/$index \\\n",
    "                        -r $working_dir/reference_genomes/$bacteria_frontTrimmed -o $path_to_output_dir \\\n",
    "                        -c $bacteria_backTrimmed --phylophlan_mode accurate \\\n",
    "                        --sample_with_n_markers $markers_per_sample --marker_in_n_samples $marker_threshold \\\n",
    "                        --nproc 8\n",
    "\n",
    "        print(\"\\n\\n\")\n",
    "    print(\"Done running all available strain trees!\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Running RAxML GTRGAMMA on s__Akkermansia_muciniphila.fna:\n",
      "Warning, you specified a working directory via \"-w\"\n",
      "Keep in mind that RAxML only accepts absolute path names, not relative ones!\n",
      "\n",
      "RAxML can't, parse the alignment file as phylip file \n",
      "it will now try to parse it as FASTA file\n",
      "\n",
      "RAxML output files with the run ID <s__Akkermansia_muciniphila.StrainPhlAn3.tre> already exist \n",
      "in directory /panfs/panfs1.ucsd.edu/panscratch/jhc103/VertMetaphlan-frmerged/strain-trees-output/output_50_A_muciniphila/GAMMA/ ...... exiting\n",
      "\n",
      "\n",
      "\n",
      "Running RAxML GTRGAMMA on s__Bacteroides_dorei.fna:\n",
      "Warning, you specified a working directory via \"-w\"\n",
      "Keep in mind that RAxML only accepts absolute path names, not relative ones!\n",
      "\n",
      "RAxML can't, parse the alignment file as phylip file \n",
      "it will now try to parse it as FASTA file\n",
      "\n",
      "RAxML output files with the run ID <s__Bacteroides_dorei.StrainPhlAn3.tre> already exist \n",
      "in directory /panfs/panfs1.ucsd.edu/panscratch/jhc103/VertMetaphlan-frmerged/strain-trees-output/output_50_B_dorei/GAMMA/ ...... exiting\n",
      "\n",
      "\n",
      "\n",
      "Running RAxML GTRGAMMA on s__Bacteroides_fragilis.fna:\n",
      "Warning, you specified a working directory via \"-w\"\n",
      "Keep in mind that RAxML only accepts absolute path names, not relative ones!\n",
      "\n",
      "RAxML can't, parse the alignment file as phylip file \n",
      "it will now try to parse it as FASTA file\n",
      "\n",
      "RAxML output files with the run ID <s__Bacteroides_fragilis.StrainPhlAn3.tre> already exist \n",
      "in directory /panfs/panfs1.ucsd.edu/panscratch/jhc103/VertMetaphlan-frmerged/strain-trees-output/output_50_B_fragilis/GAMMA/ ...... exiting\n",
      "\n",
      "\n",
      "\n",
      "Running RAxML GTRGAMMA on s__Bacteroides_thetaiotaomicron.fna:\n",
      "Warning, you specified a working directory via \"-w\"\n",
      "Keep in mind that RAxML only accepts absolute path names, not relative ones!\n",
      "\n",
      "RAxML can't, parse the alignment file as phylip file \n",
      "it will now try to parse it as FASTA file\n",
      "\n",
      "RAxML output files with the run ID <s__Bacteroides_thetaiotaomicron.StrainPhlAn3.tre> already exist \n",
      "in directory /panfs/panfs1.ucsd.edu/panscratch/jhc103/VertMetaphlan-frmerged/strain-trees-output/output_50_B_thetaiotaomicron/GAMMA/ ...... exiting\n",
      "\n",
      "\n",
      "\n",
      "Running RAxML GTRGAMMA on s__Bacteroides_uniformis.fna:\n",
      "Warning, you specified a working directory via \"-w\"\n",
      "Keep in mind that RAxML only accepts absolute path names, not relative ones!\n",
      "\n",
      "RAxML can't, parse the alignment file as phylip file \n",
      "it will now try to parse it as FASTA file\n",
      "\n",
      "RAxML output files with the run ID <s__Bacteroides_uniformis.StrainPhlAn3.tre> already exist \n",
      "in directory /panfs/panfs1.ucsd.edu/panscratch/jhc103/VertMetaphlan-frmerged/strain-trees-output/output_50_B_uniformis/GAMMA/ ...... exiting\n",
      "\n",
      "\n",
      "\n",
      "Running RAxML GTRGAMMA on s__Bacteroides_vulgatus.fna:\n",
      "Warning, you specified a working directory via \"-w\"\n",
      "Keep in mind that RAxML only accepts absolute path names, not relative ones!\n",
      "\n",
      "RAxML can't, parse the alignment file as phylip file \n",
      "it will now try to parse it as FASTA file\n",
      "\n",
      "RAxML output files with the run ID <s__Bacteroides_vulgatus.StrainPhlAn3.tre> already exist \n",
      "in directory /panfs/panfs1.ucsd.edu/panscratch/jhc103/VertMetaphlan-frmerged/strain-trees-output/output_50_B_vulgatus/GAMMA/ ...... exiting\n",
      "\n",
      "\n",
      "\n",
      "Running RAxML GTRGAMMA on s__Clostridium_perfringens.fna:\n",
      "Warning, you specified a working directory via \"-w\"\n",
      "Keep in mind that RAxML only accepts absolute path names, not relative ones!\n",
      "\n",
      "RAxML can't, parse the alignment file as phylip file \n",
      "it will now try to parse it as FASTA file\n",
      "\n",
      "RAxML output files with the run ID <s__Clostridium_perfringens.StrainPhlAn3.tre> already exist \n",
      "in directory /panfs/panfs1.ucsd.edu/panscratch/jhc103/VertMetaphlan-frmerged/strain-trees-output/output_50_C_perfringens/GAMMA/ ...... exiting\n",
      "\n",
      "\n",
      "\n",
      "Running RAxML GTRGAMMA on s__Enterococcus_faecalis.fna:\n",
      "Warning, you specified a working directory via \"-w\"\n",
      "Keep in mind that RAxML only accepts absolute path names, not relative ones!\n",
      "\n",
      "RAxML can't, parse the alignment file as phylip file \n",
      "it will now try to parse it as FASTA file\n",
      "\n",
      "RAxML output files with the run ID <s__Enterococcus_faecalis.StrainPhlAn3.tre> already exist \n",
      "in directory /panfs/panfs1.ucsd.edu/panscratch/jhc103/VertMetaphlan-frmerged/strain-trees-output/output_50_E_faecalis/GAMMA/ ...... exiting\n",
      "\n",
      "\n",
      "\n",
      "Running RAxML GTRGAMMA on s__Escherichia_coli.fna:\n",
      "Warning, you specified a working directory via \"-w\"\n",
      "Keep in mind that RAxML only accepts absolute path names, not relative ones!\n",
      "\n",
      "RAxML can't, parse the alignment file as phylip file \n",
      "it will now try to parse it as FASTA file\n",
      "\n",
      "RAxML output files with the run ID <s__Escherichia_coli.StrainPhlAn3.tre> already exist \n",
      "in directory /panfs/panfs1.ucsd.edu/panscratch/jhc103/VertMetaphlan-frmerged/strain-trees-output/output_50_E_coli/GAMMA/ ...... exiting\n",
      "\n",
      "\n",
      "\n",
      "Running RAxML GTRGAMMA on s__Lactobacillus_rodentium.fna:\n",
      "Warning, you specified a working directory via \"-w\"\n",
      "Keep in mind that RAxML only accepts absolute path names, not relative ones!\n",
      "\n",
      "RAxML can't, parse the alignment file as phylip file \n",
      "it will now try to parse it as FASTA file\n",
      "\n",
      "RAxML output files with the run ID <s__Lactobacillus_rodentium.StrainPhlAn3.tre> already exist \n",
      "in directory /panfs/panfs1.ucsd.edu/panscratch/jhc103/VertMetaphlan-frmerged/strain-trees-output/output_50_L_rodentium/GAMMA/ ...... exiting\n",
      "\n",
      "\n",
      "\n",
      "Running RAxML GTRGAMMA on s__Parabacteroides_distasonis.fna:\n",
      "Warning, you specified a working directory via \"-w\"\n",
      "Keep in mind that RAxML only accepts absolute path names, not relative ones!\n",
      "\n",
      "RAxML can't, parse the alignment file as phylip file \n",
      "it will now try to parse it as FASTA file\n",
      "\n",
      "RAxML output files with the run ID <s__Parabacteroides_distasonis.StrainPhlAn3.tre> already exist \n",
      "in directory /panfs/panfs1.ucsd.edu/panscratch/jhc103/VertMetaphlan-frmerged/strain-trees-output/output_50_P_distasonis/GAMMA/ ...... exiting\n",
      "\n",
      "\n",
      "\n",
      "Running RAxML GTRGAMMA on s__Prevotella_copri.fna:\n",
      "Warning, you specified a working directory via \"-w\"\n",
      "Keep in mind that RAxML only accepts absolute path names, not relative ones!\n",
      "\n",
      "RAxML can't, parse the alignment file as phylip file \n",
      "it will now try to parse it as FASTA file\n",
      "\n",
      "RAxML output files with the run ID <s__Prevotella_copri.StrainPhlAn3.tre> already exist \n",
      "in directory /panfs/panfs1.ucsd.edu/panscratch/jhc103/VertMetaphlan-frmerged/strain-trees-output/output_50_P_copri/GAMMA/ ...... exiting\n",
      "\n",
      "\n",
      "\n",
      "Running RAxML GTRGAMMA on s__Pseudomonas_yamanorum.fna:\n",
      "Warning, you specified a working directory via \"-w\"\n",
      "Keep in mind that RAxML only accepts absolute path names, not relative ones!\n",
      "\n",
      "RAxML can't, parse the alignment file as phylip file \n",
      "it will now try to parse it as FASTA file\n",
      "\n",
      "RAxML output files with the run ID <s__Pseudomonas_yamanorum.StrainPhlAn3.tre> already exist \n",
      "in directory /panfs/panfs1.ucsd.edu/panscratch/jhc103/VertMetaphlan-frmerged/strain-trees-output/output_50_P_yamanorum/GAMMA/ ...... exiting\n",
      "\n",
      "\n",
      "\n",
      "Done rebuilding all tree using GTRGAMMA model\n"
     ]
    }
   ],
   "source": [
    "for index, row in df_bac_marker.iterrows():\n",
    "    \n",
    "    ## Create the two bacteria name string you might want to utilize\n",
    "    bacteria_frontTrimmed = index.strip(\"s__\")\n",
    "    bacteria_backTrimmed = index.strip(\"fna\").strip(\".\")\n",
    "    bacteria_trimmed = bacteria_frontTrimmed.strip(\"fna\").strip(\".\")\n",
    "    bacteria_simplified=bacteria_trimmed[0]+bacteria_trimmed[bacteria_trimmed.find(\"_\"):len(bacteria_trimmed)]\n",
    "    \n",
    "\n",
    "#     ## marker_sample count\n",
    "    markers_per_sample = row['marker_threshold']\n",
    "    \n",
    "    # Set up output directory then create it if not created yet \n",
    "    path_to_output_dir1 = join(tree_dir, \"output_\"+marker_threshold+\"_\"+bacteria_simplified)\n",
    "    path_to_output_dir = join(path_to_output_dir1,\"GAMMA\")\n",
    "    if not os.path.isdir(path_to_output_dir):\n",
    "        os.mkdir(path_to_output_dir)\n",
    "    \n",
    "    print(\"Running RAxML GTRGAMMA on \"+ index +\":\")\n",
    "    \n",
    "    path_to_aln = join(path_to_output_dir1, bacteria_backTrimmed+\".StrainPhlAn3_concatenated.aln\")\n",
    "    output_name = bacteria_backTrimmed+\".StrainPhlAn3.tre\"\n",
    "\n",
    "    # Run RAxML with GTRGAMMA\n",
    "    ## example alignment file s__Bacteroides_vulgatus.StrainPhlAn3_concatenated.aln\n",
    "    !/home/jhc103/miniconda3/envs/mpa3-py37/bin/raxmlHPC-PTHREADS-SSE3 -p 1989 -m GTRGAMMA -T 8 -w $path_to_output_dir -s $path_to_aln -n $output_name\n",
    "\n",
    "    print(\"\\n\\n\")\n",
    "print(\"Done rebuilding all tree using GTRGAMMA model\")\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Example strainphlan run! \n",
    "strainphlan -s consensus_markers_nonMammallianHost/*.pkl -m db_markers/s__Collinsella_stercoris.fna -r reference_genomes/Collinsella_stercoris.fna -o output_20_C_stercoris_nonMammallianHost/ -c s__Collinsella_stercoris --phylophlan_mode accurate --sample_with_n_markers 5 --marker_in_n_samples 20"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "/home/jhc103/miniconda3/envs/mpa3-py37/bin/raxmlHPC-PTHREADS-SSE3 -p 1989 -m GTRGAMMA -T 8 -w /panfs/panfs1.ucsd.edu/panscratch/jhc103/VertMetaphlan-frmerged/output_20_B_vulgatus -s output_20_B_vulgatus/./s__Bacteroides_vulgatus.StrainPhlAn3_concatenated.aln -n s__Bacteroides_vulgatus.StrainPhlAn3.tre"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "!/home/jhc103/miniconda3/envs/mpa3-py37/bin/raxmlHPC-PTHREADS-SSE3 -p 1989 -m GTRGAMMA -T 8 -w /panfs/panfs1.ucsd.edu/panscratch/jhc103/VertMetaphlan-frmerged/strain-trees-output/output_65_A_muciniphila/GAMMA -s /panfs/panfs1.ucsd.edu/panscratch/jhc103/VertMetaphlan-frmerged/strain-trees-output/output_65_A_muciniphila/s__Akkermansia_muciniphila.StrainPhlAn3_concatenated.aln -n s__Akkermansia_muciniphila.StrainPhlAn3.tre"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "mpa3-py37",
   "language": "python",
   "name": "mpa3-py37"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.7.10"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
